pkg/build/strategies/sti/sti.go

```diff
@@ -250,6 +248,15 @@ func (b *STI) Determine(request *api.Request) (err error) {
 		return
 	}
 
+	// if image does not exist in local registry try searching for it
+	if !previousImageExists && request.ForcePull {
```

Comment:
You're right about the `forcePull` flag. But I'm not sure what you mean by the later. That's true I should be downloading the image even if previous exists (might be older version), but I'm not sure what you mean by 1?